Dalamud/Game/Gui/Dtr/DtrBar.cs

```diff
@@ -290,21 +368,30 @@ private void Update(IFramework unused)
 
             data.Dirty = false;
         }
+
+        this.entriesLock.ExitReadLock();
     }
 
     private void HandleAddedNodes()
     {
-        if (!this.newEntries.IsEmpty)
+        var dtrOrder = this.configuration.DtrOrder ??= [];
+
+        this.entriesLock.EnterUpgradeableReadLock();
+
+        foreach (var entry in this.entries)
         {
-            foreach (var newEntry in this.newEntries)
-            {
-                newEntry.TextNode = this.MakeNode(++this.runningNodeIds);
-                this.entries.Add(newEntry);
-            }
-            
-            this.newEntries.Clear();
-            this.ApplySort();
+            // Is the node going away already, or already has been added?
+            if (entry.ShouldBeRemoved || entry.Added)
+                continue;
+
+            // Add the entry to the end of the order list, if it's not there already.
+            if (!dtrOrder.Contains(entry.Title))
+                dtrOrder.Add(entry.Title);
         }
+
+        this.ApplySort(dtrOrder);
```

Comment:
This sort used to run only when new entries were added, but now it runs each frame. Would it be possible to run this only if a change was found in the above loop?

Reply:
Changed to sort on adding to entries.